jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -367,87 +352,77 @@ ecma_op_create_dynamic_function (const ecma_value_t *arguments_list_p, /**< argu
   ECMA_STRING_TO_UTF8_STRING (arguments_str_p, arguments_buffer_p, arguments_buffer_size);
   ECMA_STRING_TO_UTF8_STRING (function_body_str_p, function_body_buffer_p, function_body_buffer_size);
 
-#if ENABLED (JERRY_RESOURCE_NAME)
-  JERRY_CONTEXT (resource_name) = ecma_make_magic_string_value (LIT_MAGIC_STRING_RESOURCE_ANON);
-#endif /* ENABLED (JERRY_RESOURCE_NAME) */
-
-  ecma_compiled_code_t *bytecode_data_p = NULL;
+  ecma_value_t resource_name = ecma_make_magic_string_value (LIT_MAGIC_STRING_RESOURCE_ANON);
+  ecma_compiled_code_t *bytecode_p = parser_parse_script (arguments_buffer_p,
+                                                          arguments_buffer_size,
+                                                          function_body_buffer_p,
+                                                          function_body_buffer_size,
+                                                          resource_name,
+                                                          parse_opts);
 
-  ecma_value_t ret_value = parser_parse_script (arguments_buffer_p,
-                                                arguments_buffer_size,
-                                                function_body_buffer_p,
-                                                function_body_buffer_size,
-                                                parse_opts,
-                                                &bytecode_data_p);
+  ECMA_FINALIZE_UTF8_STRING (function_body_buffer_p, function_body_buffer_size);
+  ECMA_FINALIZE_UTF8_STRING (arguments_buffer_p, arguments_buffer_size);
+  ecma_deref_ecma_string (arguments_str_p);
+  ecma_deref_ecma_string (function_body_str_p);
 
-  if (!ECMA_IS_VALUE_ERROR (ret_value))
+  if (JERRY_UNLIKELY (bytecode_p == NULL))
   {
-    JERRY_ASSERT (ecma_is_value_true (ret_value));
+    return ECMA_VALUE_ERROR;
+  }
 
 #if ENABLED (JERRY_ESNEXT)
-    ecma_value_t *func_name_p;
-    func_name_p = ecma_compiled_code_resolve_function_name ((const ecma_compiled_code_t *) bytecode_data_p);
-    *func_name_p = ecma_make_magic_string_value (LIT_MAGIC_STRING_ANONYMOUS);
+  ecma_value_t *func_name_p;
+  func_name_p = ecma_compiled_code_resolve_function_name ((const ecma_compiled_code_t *) bytecode_p);
+  *func_name_p = ecma_make_magic_string_value (LIT_MAGIC_STRING_ANONYMOUS);
```

Comment:
What is the difference between LIT_MAGIC_STRING_RESOURCE_ANON and LIT_MAGIC_STRING_ANONYMOUS? Why the former uses abbreviation?

Reply:
LIT_MAGIC_STRING_RESOURCE_ANON is '<anonymous>', and is used in resource names, while LIT_MAGIC_STRING_ANONYMOUS is 'anonymous' and is used in function names.

Reply:
So it is the same as empty? Perhaps we could use a macro alias then?

Reply:
Sorry, it's not empty, its `'<anonymous>'` (github thought it was markdown syntax, and removed it).